mlir-sycl/include/mlir/Dialect/SYCL/Analysis/ConstructorBaseAnalysis.h

```diff
@@ -0,0 +1,158 @@
+//===- ConstructorBaseAnalysis.h - Base analysis for SYCL types -*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Base class for analyses that determine properties of instances of SYCL types
+// based on their construction.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_SYCL_ANALYSIS_CONSTRUCTORBASEANALYSIS_H
+#define MLIR_DIALECT_SYCL_ANALYSIS_CONSTRUCTORBASEANALYSIS_H
+
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/Polygeist/Analysis/DataFlowSolverWrapper.h"
+#include "mlir/Dialect/Polygeist/Analysis/ReachingDefinitionAnalysis.h"
+#include "mlir/Dialect/SYCL/Analysis/AliasAnalysis.h"
+#include "mlir/Pass/AnalysisManager.h"
```

Comment:
Can we include `LLVMTypes.h` instead of the whole dialect? That file is **huge**.